inbm/dispatcher-agent/dispatcher/schedule/schedules.py

```diff
@@ -9,17 +9,22 @@
 from typing import Optional
 
 @dataclass(init=True)
-class SingleSchedule:
-    """ Represents a SingleSchedule object from the Scheduled Manifest """
+class Schedule:
+    """ Represents a Base class for schedule objects """
     request_id: str
+    id: Optional[int] = field(default=None)    
+
+
+@dataclass(init=True)
+class SingleSchedule(Schedule):
+    """ Represents a SingleSchedule object from the Scheduled Manifest """ 
     start_time: Optional[datetime] = field(default=None)
     end_time: Optional[datetime] = field(default=None)
     manifests: list[str] = field(default_factory=list)
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Refactoring$</h2>**
**Suggestion:** Ensure SingleSchedule and RepeatedSchedule have a method to check if a given datetime matches their schedule.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    from datetime import datetime

    @dataclass
    class SingleSchedule(Schedule):
        def is_active(self, check_time: datetime) -> bool:
            return self.start_time <= check_time <= self.end_time

    @dataclass
    class RepeatedSchedule(Schedule):
        def is_active(self, check_time: datetime) -> bool:
            # Assuming a method exists to evaluate this based on cron expression
            return True
```